book/src/libraries.md

```diff
@@ -1,7 +1,131 @@
+
+Each [POU](pous.md), [Global Variable](variables.md), or [Datatype](datatypes.md) defined in the included files will be added to the project.
+POUs and Global variables included with the `-i` are marked as external, the implementation part of a POU is ignored.
+
+To link the library, two options are then available:
+
+- <a name="shared-libraries">**Shared Libraries** </a>
+
+    A shared library (i.e. extension `.so`) can be linked using the `-l` flag. </br>
+    For a library called `mylib`, when the flag `-lmylib` is passed, the linker will search for a file called `libmylib.so`.
+    > **Note that the `lib<LibName>.so` format is required by the linker for unix like systems**
+
+    The library locations used by the linker are the default search locations of the linker (i.e. `/usr/lib`, `/lib`), additional pathes can be provided using the `-L` flag (e.g `-L/opt/lib` will make the linker also search for files in /opt/lib).</br>
+    Additionally, the environment variable `LD_LIBRARY_PATH` can be defined to append entries to the linker's search location.
+
+- <a name="static-libraries">**Static Libraries**</a>
+
+    Static libraries compiled as object files can be linked by simply passing the object file (i.e. extension `.o`) as an input (simlar to other `.st` files).
+
+    Archive files (i.e. extension `.a`) can be linked similarly to [Shared Libraries](#shared-libraries) using the `-l` flag. 
+    If the application is being compiled with the `--static` flag (or no shared library (`.so`) is found), the linker will use the archive file.</br>
+
+    > If neither a shared object (`.so`) or an archive file (`.a`) is found, compilation will fail.
+
+### Command line example
+
+To compile a file called `input.st` including a header and linking a library called `libiec.so` from `/lib` : 
+
+```sh
+rustyc input.st -i iec/header.st -L/lib/ -liec
+```
+
+
+## Linking libraries using the Build Description File `plc.json`
+
+Libraries can be added to a project managed with a [Build Description File](using_rusty/build_description_file.md#build-description-file-plcjson). </br>
+To add a library to the project, the `"libraries"` section can be used.
+A library entry requires a `name`, a `path`, the `package` behaviour, and a set of files to include (`include_path`).
+
+### `name`
+
+The name of the library to be linked. This will be used by the linker to find the library. </br>
+A library with the name `mylib` must have an equivalant compiled file called `libmylib.so`.
+> Note, archive files (ending with `.a`) are currently not supported.
+
+### `path`
+
+The location of the library to be linked. The path can be either absolute or relative to the project. </br>
+
+### `package`
+
+The packaging option for the library, i.e wether the library should be copied or is already available on the system.</br>
+The value <a name="copy">`"Copy"`</a> indicates that the given library should be copied to the [Library Location](#library-location). </br>
```

Comment:
Copy and System are anchors? they look like links in the resulting book but go nowhere? have we done this somewhere else?

Reply:
Initially, they were <h> tags, but they didn't look good, i still wanted to link to them.